h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -130,15 +131,15 @@
 
   /** Incoming TCP request.  Make a read-mode AutoBuffer from the open Channel,
    *  figure the originating H2ONode from the first few bytes read. */
-  AutoBuffer( SocketChannel sock ) throws IOException {
+  AutoBuffer( ByteChannel sock, InetAddress address  ) throws IOException {
```

Comment:
This constructor is used in receiver threads so it's the inet of the receiving socket. I split it into 2 arguments because the SSL socket channel is extending ByteChannel and not SocketChannel (this was much easier to extend for me), but the way I'm getting the InetAddress is the same as in the old code:

https://github.com/h2oai/h2o-3/blob/739ec856995d066ccecb8eb605ca9ef5a9d3baa6/h2o-core/src/main/java/water/TCPReceiverThread.java#L68

https://github.com/h2oai/h2o-3/blob/d4f05e3a6a099d277494b30611ac627e9715e5c2/h2o-core/src/main/java/water/AutoBuffer.java#L141